src/EditorFeatures/Next/Structure/BlockContextProvider.cs

```diff
@@ -50,23 +73,119 @@ public void Dispose()
 
         private class RoslynBlockContext : ForegroundThreadAffinitizedObject, IBlockContext
         {
+            private readonly RoslynBlockContextProvider _provider;
+
             public IBlockTag BlockTag { get; }
 
             public ITextView TextView { get; }
 
-            public RoslynBlockContext(IBlockTag blockTag, ITextView textView)
+            public RoslynBlockContext(
+                RoslynBlockContextProvider provider,
+                IBlockTag blockTag,
+                ITextView textView)
             {
+                _provider = provider;
                 BlockTag = blockTag;
                 TextView = textView;
             }
 
-            public object Content
+            public object Content => CreateContent();
+
+            private object CreateContent()
             {
-                get
+                this.AssertIsForeground();
+
+                var result = new ViewHostingControl(
+                    CreateElisionBufferView, CreateProjectionBufferForBlockHeaders);
+
+                return result;
+            }
+
+            private IWpfTextView CreateElisionBufferView(ITextBuffer finalBuffer)
+            {
+                return RoslynOutliningRegionTag.CreateShrunkenTextView(
+                    _provider._textEditorFactoryService, finalBuffer);
+            }
+
+            private ITextBuffer CreateProjectionBufferForBlockHeaders()
```

Comment:
No.  But i will attempt to add one.  In general, we're not great with very high level UI tests.